### PR TITLE
fix: loosing data when moving between steps on the checkout

### PR DESCRIPTION
**Pull request by KevinGorjan**
This ticket fixes 
* #410

## Description
The user can't navigate to any next page anymore, this can only be done when the form from the current step correctly is filled in.
If the user goes directly to the page by URL, the application will check of the previous step is valid, else the user will be redirected to the user-account page

## Related Issue
* #409  

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/pages/Checkout/UserAccount.vue [hunk 1]
```diff
@@ -251,14 +253,23 @@ export default defineComponent({
...
         form.value.firstname = user.value.firstname;
         form.value.lastname = user.value.lastname;
         form.value.email = user.value.email;
       }
     });
 
+    onMounted(async () => {
+      const checkout = await getItem('checkout');
+      if (checkout && checkout['user-account']) {
+        const data = checkout['user-account'];
```
Comment (Frodigo) [inline packages/theme/pages/Checkout/UserAccount.vue:266]: I didn't see the logic for filling data in other steps. It's possible to do something like this in other steps as well?

**Review by Frodigo — CHANGES_REQUESTED** [review]

File: packages/theme/pages/Checkout/UserAccount.vue (hunk 1 above)
Comment (KevinGorjan) [reply packages/theme/pages/Checkout/UserAccount.vue:266]: On the shipping and billing step, the form gets the data when fetching the cart.

image

That's why I don't store it in the localstorage.

File: packages/theme/pages/Checkout/Payment.vue [hunk 2]
```diff
@@ -209,11 +211,20 @@ export default defineComponent({
       await load();
     });
 
+    onMounted(async () => {
+      const checkout = await getItem('checkout');
+      if (!checkout || !checkout.billing) {
```
[the remark this answers is not in the document]
Comment (KevinGorjan) [reply packages/theme/pages/Checkout/Payment.vue:216]: Good idea, I'll add this ;-)

**Comment by KevinGorjan** [comment]
@Frodigo I added tests for the asyncLocalStorage and the steps-helper for the checkout.

I didn't add tests in the pages because this will be done in a separate ticket

**Review by Frodigo — APPROVED** [review]

**Pull request merged**